### ISSUE TITLE
Error with mistune

### ISSUE THREAD

**Issue by dattaz** [openzim/sotoki#108]
Apparently, sometime mistune, which do markdown to html crash.
We should investigate and fix this issue.

Step to reproduice :
```
sotoki wordpress.stackexchange.com Kiwix --threads=1 --reset --clean-previous --no-userprofile --nopic --nozim
```

```
  File "/home/dattaz/dev/kiwix/sotoki/sotoki/venv3/bin/sotoki", line 11, in <module>
    load_entry_point('sotoki==1.2', 'console_scripts', 'sotoki')()
  File "/home/dattaz/dev/kiwix/sotoki/sotoki/venv3/lib/python3.7/site-packages/sotoki-1.2-py3.7.egg/sotoki/sotoki.py", line 1530, in run
    parser.parse(os.path.join(dump, "prepare.xml"))
  File "/usr/lib/python3.7/xml/sax/expatreader.py", line 111, in parse
    xmlreader.IncrementalParser.parse(self, source)
  File "/usr/lib/python3.7/xml/sax/xmlreader.py", line 125, in parse
    self.feed(buffer)
  File "/usr/lib/python3.7/xml/sax/expatreader.py", line 217, in feed
    self._parser.Parse(data, isFinal)
  File "../Modules/pyexpat.c", line 417, in StartElement
  File "/usr/lib/python3.7/xml/sax/expatreader.py", line 333, in start_element
    self._cont_handler.startElement(name, AttributesImpl(attrs))
  File "/home/dattaz/dev/kiwix/sotoki/sotoki/venv3/lib/python3.7/site-packages/sotoki-1.2-py3.7.egg/sotoki/sotoki.py", line 199, in startElement
    tmp["Text"] = markdown(html.escape(tmp["Text"], quote=False))
  File "/home/dattaz/dev/kiwix/sotoki/sotoki/venv3/lib/python3.7/site-packages/sotoki-1.2-py3.7.egg/sotoki/sotoki.py", line 767, in markdown
    text_html = MARKDOWN(text)[3:-5]
  File "/home/dattaz/dev/kiwix/sotoki/sotoki/venv3/lib/python3.7/site-packages/mistune-2.0.0a2-py3.7.egg/mistune/markdown.py", line 69, in __call__
    return self.parse(s)
  File "/home/dattaz/dev/kiwix/sotoki/sotoki/venv3/lib/python3.7/site-packages/mistune-2.0.0a2-py3.7.egg/mistune/markdown.py", line 54, in parse
    result = self.block.render(tokens, self.inline, state)
  File "/home/dattaz/dev/kiwix/sotoki/sotoki/venv3/lib/python3.7/site-packages/mistune-2.0.0a2-py3.7.egg/mistune/block_parser.py", line 246, in render
    return ''.join(data)
  File "/home/dattaz/dev/kiwix/sotoki/sotoki/venv3/lib/python3.7/site-packages/mistune-2.0.0a2-py3.7.egg/mistune/block_parser.py", line 263, in _iter_render
    yield method(children, *params)
  File "/home/dattaz/dev/kiwix/sotoki/sotoki/venv3/lib/python3.7/site-packages/mistune-2.0.0a2-py3.7.egg/mistune/renderers.py", line 175, in block_code
    lang = info.strip().split(None, 1)[0]
IndexError: list index out of range
```

**Comment by kelson42** [comment]
@satyamtg Would you be able to look at this bug left before we make a new release?

**Comment by satyamtg** [comment]
> @satyamtg Would you be able to look at this bug left before we make a new release?

I'm actually having a look at it presently.

**Comment by satyamtg** [comment]
This was apparently due to trailing spaces after the lines in some posts. Although mistune should handle it, but it doesn't seem to do so. I have raised an issue with them [here](https://github.com/lepture/mistune/issues/231). However, till that issue is not fixed by mistune devs, I have implemented code in the PR to strip trailing spaces from the lines of the posts.
Once the bug is fixed by mistune, we can remove that code from sotoki. But for now, it's the only solution as far as I know.
